src/components/custom-field-input-text/custom-field-input-text.test.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import CustomFieldInputText from './custom-field-input-text.jsx';
+
+describe('CustomFieldInputText', () => {
+  it('has defaults', () => {
+    const tree = renderer.create((
+      <CustomFieldInputText />
+    )).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  describe('className API', () => {
+    it('prioritizes className prop', () => {
+      const tree = renderer.create((
+        <CustomFieldInputText className="prioritize-me" />
+      )).toJSON();
+      expect(tree.props.className).toEqual('prioritize-me');
+    });
+  });
+
+  describe('disabled API', () => {
+    it('can be disabled', () => {
+      const tree = renderer.create((
+        <CustomFieldInputText disabled={true} />
+      )).root;
+      expect(tree.findByProps({ disabled: true })).toBeTruthy();
```

Comment:
What's the difference between using a truthy assertion and a specific assertion on the disabled property? Is one better than the other or are they the same?

We have historically preferred simple (and informational) error messages on test failures. When a truthy statement like this fails, it is either "expected false to be true" or "expected <large object serialization> to be true".  In this case, it is the latter.

Reply:
This is also interesting: if I were to move the `disabled={props.disabled}` form the `<input>` to the `<label>`, this test still passes. Is that intended or a reasonable pattern?

Reply:
Turns out there were a few issues with these tests! Truthiness expectations will be removed,  and all tests now explicitly `findByType` their target, turns out that `tree` contains `CustomFieldInputText` as the top level element, so all those expectations found the props on the root element, right where we put them a line before in the `renderer.create` call. 😆